src/shared/components/web3-auth.tsx

```diff
@@ -0,0 +1,18 @@
+'use client';
+
+import { UserContext } from '@/context/user';
+import { useContext, useEffect } from 'react';
+import { useAccount } from 'wagmi';
+
+export const Web3Auth = ({ children }) => {
+    const { user, authUser } = useContext(UserContext);
+    const { address } = useAccount();
+
+    useEffect(() => {
+        if (address && !user) {
+            authUser();
+        }
+    }, [address]);
```

Comment:
a few deps missing here. we need to review our lint setup. 
Should not affect us in any significant way though.